lib/iris/analysis/maths.py

```diff
@@ -336,17 +387,12 @@ def _inplace_common_checks(cube, other, math_op):
     cast back to the integer data of `cube`.
 
     """
-    if cube.dtype.kind in 'iu':
-        if hasattr(other, 'dtype'):
-            other_kind = other.dtype.kind
-        else:
-            other_kind = np.asanyarray(other).dtype.kind
-        if other_kind not in 'iu':
-            # Cannot coerce math op between int `cube` and float `other`
-            # back to int.
+    other_dtype = _get_dtype(other)
+    if not np.can_cast(other_dtype, cube.dtype, 'same_kind'):
```

Comment:
I've modified this method to make it more general, and fix an issue where trying to do in-place operations on a cube with unsigned data, and signed data would fail silently.